swarm/fuse/swarmfs_test.go

```diff
@@ -211,790 +251,1445 @@ type testAPI struct {
 
 	// mount nonexistent hash
-	_, err = swarmfs.Mount("0xfea11223344", testMountDir1)
+	failDir, err3 := addDir(dat.testDir, "remount-fail")
+	if err3 != nil {
+		t.Fatalf("Error creating remount dir: %v", bzzHash2)
+	}
+	failHash := "0xfea11223344"
+	_, err = swarmfs.Mount(failHash, failDir)
 	if err == nil {
-		t.Fatalf("Error mounting hash  %v", bzzHash2)
+		t.Fatalf("Expected this mount to fail due to non existing hash. But succeeded %v", failHash)
 	}
+	log.Debug("Nonexistent hash hasn't been mounted. Correct.")
 }
 
 func (ta *testAPI) unmountEncrypted(t *testing.T) {
+	log.Debug("Starting unmountEncrypted test")
 	ta.unmount(t, true)
+	log.Debug("Test unmountEncrypted terminated")
 }
 
 func (ta *testAPI) unmountNonEncrypted(t *testing.T) {
+	log.Debug("Starting unmountNonEncrypted test")
 	ta.unmount(t, false)
+	log.Debug("Test unmountNonEncrypted terminated")
 }
 
+//mount then unmount and check that it has been unmounted
 func (ta *testAPI) unmount(t *testing.T, toEncrypt bool) {
-	files := make(map[string]fileInfo)
-	uploadDir, _ := ioutil.TempDir(os.TempDir(), "ex-upload")
-	testMountDir, _ := ioutil.TempDir(os.TempDir(), "ex-mount")
+	dat, err := ta.initSubtest("unmount")
+	if err != nil {
+		t.Fatalf("Couldn't initialize subtest dirs: %v", err)
+	}
+	defer os.RemoveAll(dat.testDir)
 
-	files["1.txt"] = fileInfo{0700, 333, 444, getRandomBytes(10)}
-	bzzHash := createTestFilesAndUploadToSwarm(t, ta.api, files, uploadDir, toEncrypt)
+	dat.toEncrypt = toEncrypt
+	dat.testUploadDir = filepath.Join(dat.testDir, "ex-upload1")
+	dat.testMountDir = filepath.Join(dat.testDir, "ex-mount1")
+	dat.files = make(map[string]fileInfo)
+	dat.files["1.txt"] = fileInfo{0700, 333, 444, getRandomBytes(10)}
 
-	swarmfs := mountDir(t, ta.api, files, bzzHash, testMountDir)
-	defer swarmfs.Stop()
+	dat, err = ta.uploadAndMount(dat, t)
+	if err != nil {
+		t.Fatalf("Error during upload of files to swarm / mount of swarm dir: %v", err)
+	}
+	defer dat.swarmfs.Stop()
 
-	swarmfs.Unmount(testMountDir)
+	_, err = dat.swarmfs.Unmount(dat.testMountDir)
+	if err != nil {
+		t.Fatalf("could not unmount  %v", dat.bzzHash)
+	}
+	log.Debug("Unmounted Dir")
 
 	mi := swarmfs.Listmounts()
+	log.Debug("Going to list mounts")
 	for _, minfo := range mi {
-		if minfo.MountPoint == testMountDir {
-			t.Fatalf("mount state not cleaned up in unmount case %v", testMountDir)
+		log.Debug("Mount point in list: ", "point", minfo.MountPoint)
+		if minfo.MountPoint == dat.testMountDir {
+			t.Fatalf("mount state not cleaned up in unmount case %v", dat.testMountDir)
 		}
 	}
+	log.Debug("subtest terminated")
 }
 
 func (ta *testAPI) unmountWhenResourceBusyEncrypted(t *testing.T) {
+	log.Debug("Starting unmountWhenResourceBusyEncrypted test")
 	ta.unmountWhenResourceBusy(t, true)
+	log.Debug("Test unmountWhenResourceBusyEncrypted terminated")
 }
 func (ta *testAPI) unmountWhenResourceBusyNonEncrypted(t *testing.T) {
+	log.Debug("Starting unmountWhenResourceBusyNonEncrypted test")
```

Comment:
This test on macOS. Internal unmount https://github.com/ethersphere/go-ethereum/blob/swarm-network-rewrite/swarm/fuse/swarmfs_unix.go#L202 fails with an error unmount /var/folders/l4/tn9ytbgs5xx76lshwgx5bj1w0000gn/T/unmountWhenResourceBusy174427739/ex-mount1: resource busy, but there is a fallback strategy to use external unmount https://github.com/ethersphere/go-ethereum/blob/swarm-network-rewrite/swarm/fuse/swarmfs_util.go#L41 and in case of macOS, diskutil umount uses force option. This is a bit dangerous feature, as data may be lost in this case. There are workarounds for linux, to lazy unmount a fuse or ru use umount command instead of fusermount. But, if the "force" argument is removed from fuse.externalUnmount function, the test passes on macOS, too.